packages/@lwc/style-compiler/src/scope-at-rules/transform.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { Root } from 'postcss';
+import { SHADOW_ATTRIBUTE } from '../utils/selectors-scoping';
+
+export default function process(root: Root) {
+    const knownNames: Set<string> = new Set();
+    root.walkAtRules((atRule) => {
+        if (atRule.name === 'keyframes') {
+            const { params } = atRule;
+            knownNames.add(params);
+            atRule.params = `${params}-${SHADOW_ATTRIBUTE}`;
+        }
+    });
+    root.walkRules((rule) => {
+        rule.walkDecls((decl) => {
+            if (decl.prop === 'animation') {
+                // Use a simple heuristic of breaking up the tokens by whitespace. We could use
+                // a dedicated animation prop parser (e.g.
+                // https://github.com/hookhookun/parse-animation-shorthand) but it's
+                // probably overkill.
+                const tokens = decl.value
+                    .trim()
+                    .split(/\s+/g)
+                    .map((token) =>
+                        knownNames.has(token) ? `${token}-${SHADOW_ATTRIBUTE}` : token
```

Comment:
@abdulsattar That's a good point. Sadly I found a bug in the library related to `var()` expressions, which is partly why I didn't go with it (https://github.com/hookhookun/parse-animation-shorthand/issues/10).

 I could use the library and then `try`/`catch`, and if it throws an error, fall back to `String.split()`? What do you think?

Reply:
Alternatively, we could use something like https://github.com/TrySound/postcss-value-parser but then we have to reimplement some of the logic to understand which part of the tokens array is actually the animation name.

Reply:
Tbh, I don't think it's worth the effort. Since this is a breaking change anyway, we can say that from this point onwards, animation names can't be any of the other shorthand properties (if they're used within the same declaration).

Reply:
OK, in that case let's open an issue when this is merged and just consider it a known issue for now. It'd be nice if, in the future, we didn't have this limitation.

Reply:
Interesting; [here is a project](https://github.com/retyui/postcss-animations/blob/186f92c05d683a3ba88d35c177269edc41a6295f/src/index.js#L7-L26) that uses `postcss-values-parser` but keeps a list of known keywords to ignore.

Reply:
You know what, let's just leave it as-is. If you name your `@keyframes` as `linear` and then expect something like `animation: 1s linear linear;` to work properly, when you're probably going to have a bad time.